### PR TITLE
OSSM-1228 Reduce differences between Maistra and upstream CNI

### PR DESCRIPTION
**Pull request by luksa**

### CODE REVIEW

**Comment by luksa** [comment]
/hold

This includes the commit from https://github.com/maistra/istio-operator/pull/903, which must be merged first.

File: resources/helm/overlays/istio_cni/templates/clusterrolebinding.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{{ if .Values.enabled }}
+{{ if .Values.cni.enabled }}
```
Comment (jwendell) [inline resources/helm/overlays/istio_cni/templates/clusterrolebinding.yaml:1]: Do we still need an overlay file? Can't we just use upstream's from istio repo?

Comment (luksa) [reply resources/helm/overlays/istio_cni/templates/clusterrolebinding.yaml:1]: We do. We run multiple containers in the istio-cni-node Pods (one for each CNI version).

File: pkg/controller/common/cni/cni.go [hunk 2]
```diff
@@ -33,15 +36,17 @@ func InitConfig(m manager.Manager) (Config, error) {
 	} else {
 		log.Info(fmt.Sprintf("CNI is enabled for this installation: %v", config.Enabled))
 	}
+
+	config.Enabled = true
```
Comment (jwendell) [inline pkg/controller/common/cni/cni.go]: do we want to enable CNI even if multus is not in use?

Comment (luksa) [reply pkg/controller/common/cni/cni.go]: Yes. Multus simply allows us to support multiple Istio CNI versions at the same time (and fixes the race conditions).

**Review by rcernich — APPROVED** [review]

File: pkg/controller/servicemesh/controlplane/integration_test.go [hunk 3]
```diff
@@ -75,7 +75,7 @@ func TestBootstrapping(t *testing.T) {
 		operatorNamespace     = "istio-operator"
 		controlPlaneNamespace = "test"
 		smcpName              = "test"
-		cniDaemonSetName      = "istio-node"
+		cniDaemonSetName      = "istio-cni-node"
```
Comment (jwendell) [inline pkg/controller/servicemesh/controlplane/integration_test.go:78]: that needs docs update here: https://docs.openshift.com/container-platform/4.10/service_mesh/v2x/removing-ossm.html

Comment (luksa) [reply pkg/controller/servicemesh/controlplane/integration_test.go:78]: Thanks. Created https://issues.redhat.com/browse/OSSMDOC-574

**Pull request merged**